src/pr-info.ts

```diff
@@ -335,32 +335,28 @@ function getLastMaintainerBlessingDate(timelineItems: PR_repository_pullRequest_
     return undefined;
 }
 
-type FileLocation = ({
-    kind: "test",
-    package: string
-} | {
-    kind: "definition",
-    package: string
-} | {
-    kind: "package-meta",
-    package: string
-} | {
-    kind: "infrastructure"
-}) & { filePath: string };
-
-function categorizeFile(filePath: string): FileLocation {
-    // https://regex101.com/r/OKUe4Q/1/
-    const match = /^types\/(.*?)\/.*?[^\/](?:\.(d\.ts|tsx?))?$/.exec(filePath);
-    if (!match) return { filePath, kind: "infrastructure" };
+type FileKind = "test" | "definition" | "markdown" | "package-meta" | "infrastructure";
+
+type FileInfo = {
+    path: string,
+    kind: FileKind,
+    package: string | undefined
+};
+
+function categorizeFile(path: string): FileInfo {
+    // https://regex101.com/r/eFvtrz/1
+    const match = /^types\/(.*?)\/.*?[^\/](?:\.(d\.ts|tsx?|md))?$/.exec(path);
+    if (!match) return { path, kind: "infrastructure", package: undefined };
     const [pkg, suffix] = match.slice(1); // `suffix` can be null
-    switch ((suffix || "").toLowerCase()) {
-        case "d.ts": return { filePath, kind: "definition", package: pkg };
-        case "ts": case "tsx": return { filePath, kind: "test", package: pkg };
-        default: return { filePath, kind: "package-meta", package: pkg };
+    switch ((suffix || "")) {
```

Comment:
Did you want to lose the `.toLowerCase`?

Reply:
Yes: since the regexp is not case-insensitive anyway, so it didn't do much.  I considered adding an `i` flag, but looking at the repo there was just one file with an uppercase `.MD` which I renamed earlier, so no reason to allow weird casing now...